pyne/mcnp.py

```diff
@@ -1990,210 +1987,235 @@ def _read_meshtal_head(self, f):
+            Filestream of the meshtal file.
+    
+        Returns
+        -------
+        column_idx : dict
+            Dictionary of the column index.
+        """
 
         # skip blank line between enery bin boundaries and table headings
         f.readline()
 
-    def _read_column_order(self, f):
-        """Create dictionary with table headings as keys and their column
-        location as values. Dictionary is the private attribute _column_idx.
-        """
         line = f.readline()
         column_names = line.replace('Rel ', 'Rel_').replace(
             'Rslt * ', 'Rslt_*_').strip().split()
-        self._column_idx = dict(zip(column_names, range(0, len(column_names))))
+        column_idx = dict(zip(column_names, range(0, len(column_names))))
+        return column_idx
 
-    def _create_mesh(self, f, mesh_has_mats):
-        """Instantiate a Mesh object and tag the PyMOAB core instance
-           with results and relative errors.
+    def read_tally_results_rel_error(self, f, num_e_groups, num_ves,
+            column_idx):
         """
+        Read meshtally results and reletive error data.
+    
+        Parameters
+        ----------
+        f : str or filestream
+            Filestream of the meshtal file.
+        num_e_groups: int
+            Number of energy groups.
+        num_ves: int
+            Number of volume elements.
+        column_idx : dict
+            Dictionary of the column index.
 
-        mats = () if mesh_has_mats is True else None
-        super(MeshTally, self).__init__(structured_coords=[self.x_bounds,
-                                                           self.y_bounds, self.z_bounds],
-                                        structured=True, mats=mats)
-
-        num_vol_elements = (len(self.x_bounds)-1) * (len(self.y_bounds)-1)\
-            * (len(self.z_bounds)-1)
-        num_e_groups = len(self.e_bounds)-1
+        Returns
+        -------
+        results : numpy array
+            Tally results data, shape=(num_ves, num_e_groups)
+        rel_error : numpy array
+            Tally relative error data, shape=(num_ves, num_e_groups)
+        res_tot : numpy array
+            Tally total flux result.
+        rel_err_tot : numpy array
+            Total relative error fo flux.
+        """
 
         # get result and relative error data from file
-        result = np.empty(shape=(num_e_groups, num_vol_elements))
-        rel_error = np.empty(shape=(num_e_groups, num_vol_elements))
+        result = np.empty(shape=(num_e_groups, num_ves))
+        rel_error = np.empty(shape=(num_e_groups, num_ves))
         for i in range(0, num_e_groups):
             result_row = []
             rel_error_row = []
-            for j in range(0, num_vol_elements):
+            for j in range(0, num_ves):
                 line = f.readline().split()
-                result_row.append(float(line[self._column_idx["Result"]]))
+                result_row.append(float(line[column_idx["Result"]]))
```

Comment:
I thought `column_idx` was being stored on the class? But here it's passed in.

Reply:
Yes, `column_idx` should be stored on the class. Modified in https://github.com/pyne/pyne/pull/1199/commits/c2416ec0260a6d04d7a3eb22d696b2160ec8b5ad.